docs/src/adr/020adr-arenas.md

```diff
@@ -0,0 +1,328 @@
+---
+authors: Igor Konnov
+last revised: 2022-05-16
+---
+
+# ADR-020: Improving membership in arenas
+
+**Table of Contents**
+
+- [Summary (Overview)](#summary)
+- [Context (Problem)](#context)
+- [Options (Alternatives)](#options)
+- [Solution (Decision)](#solution)
+- [Consequences (Retrospection)](#consequences)
+
+## Summary
+
+We discuss an extension of membership edges in arenas. The main application of
```

Comment:
I think the way this is framed as an extension is misleading.
Isn't it more about lifting points-to information from the SMT encoding / SMT membership constants into the rewriter?

Reply:
Yes, arenas are also some form of static points-to information, though it is not as precise, as it could be

Reply:
I see. I didn't really get this from reading the summary/context. Maybe we can add something along the lines of

1. Static points-to information currently present in the SMT encoding is not as precise as it could be.
2. We propose to improve this by providing more accurate information on the level of arenas.
3. We achieve this via an extension of membership edges proposed here.

to the context section?